src/plugins/BuiltinPlugin.js

```diff
@@ -164,6 +164,66 @@ export const regexRules = [
         regexps: [ "(?<match>\\{[\\w_.$0-9]+\\}\\s*%)(\\s|$)" ],
         link: "https://github.com/ilib-js/i18nlint/blob/main/docs/source-no-manual-percentage-formatting.md",
         severity: "warning"
+    },
+    {
+        type: "resource-source",
+        name: "source-no-noun-replacement-params",
+        description: "Ensure that source strings do not contain replacement parameters that are nouns or adjectives.",
+        note: "Do not substitute nouns into UI strings. Use separate strings for each noun instead.",
+        regexps: [ "\\b(?<match>([Aa][Nn]?|[Tt][Hh][Ee])\\s+\\{.*?\\})" ],
+        link: "https://github.com/ilib-js/i18nlint/blob/main/docs/source-no-noun-replacement-params.md",
+        severity: "error",
+        useStripped: false
+    },
+    {
+        type: "resource-source",
+        name: "source-no-manual-currency-formatting",
+        description: "Ensure that source strings do not contain currency formatting. Currencies should be formatted using a locale-sensitive number formatter instead.",
+        note: "Do not format currencies in English strings. Use a locale-sensitive number formatter and substitute the result of that into this string.",
+        regexps: [ "(?<match>\\$\\s*\\{[\\w_.$0-9]+\\})" ],
+        link: "https://github.com/ilib-js/i18nlint/blob/main/docs/source-no-manual-currency-formatting.md",
+        severity: "error"
+    },
+    {
+        type: "resource-source",
+        name: "source-no-manual-date-formatting",
+        description: "Ensure that source strings do not contain manually formatted dates or times. Dates and times should be formatted using a locale-sensitive date formatter instead.",
+        note: "Do not format dates or times in English strings. Use a locale-sensitive date formatter and substitute the result of that into this string.",
+        regexps: [
+            "\\{years?\\}[/\\-\\. ]\\{months?\\}[/\\-\\. ]\\{days?\\}",
```

Comment:
To increase readability (mostly get rid of all those extra `\`), we could do it like
```js
/\{years?\}[/\-\. ]\{months?\}[/\-\. ]\{days?\}/.source
```

Reply:
Or update declarative rule loading to accept `string | RegExp`

Reply:
We should do that in a separate PR. This PR is about the date formatting.